packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java

```diff
@@ -53,15 +53,32 @@ void unlockInputConnection() {
     imm.restartInput(containerView);
   }
 
-  /** Creates an InputConnection from the IME thread when needed. */
+  /**
+   * Creates an InputConnection from the IME thread when needed.
+   *
+   * <p>We only need to create a {@link ThreadedInputConnectionProxy} and create an
+   * InputConnectionProxy on the IME thread when WebView is doing the same thing. So we rely on the
+   * system calling this method for WebView's proxy view in order to know when we need to create our
+   * own.
+   *
+   * <p>This method would normally be called for any View that used the InputMethodManager. We rely
+   * on flutter/engine filtering the calls we receive down to the ones in our hierarchy and the
+   * system WebView in order to know whether or not the system WebView expects an InputConnection on
+   * the IME thread.
+   */
   @Override
   public boolean checkInputConnectionProxy(final View view) {
+    // Check to see if the view param is WebView's ThreadedInputConnectionProxyView.
     View previousProxy = threadedInputConnectionProxyView;
     threadedInputConnectionProxyView = view;
     if (previousProxy == view) {
+      // This isn't ThreadedInputConnectionProxyView. Ignore it.
```

Comment:
It might still be the ThreadedInputConnectionProxyView, only that it's the same one that we've already set up the proxying for.